src/ore/src/lgbytes.rs

```diff
@@ -170,71 +202,88 @@ impl LgBytesMetrics {
-                free_count: free_count.with_label_values(&["heap"]),
-                free_capacity_bytes: free_capacity_bytes.with_label_values(&["heap"]),
+                alloc_count: alloc_count.with_label_values(&[name, "heap"]),
+                alloc_capacity_bytes: alloc_capacity_bytes.with_label_values(&[name, "heap"]),
+                free_count: free_count.with_label_values(&[name, "heap"]),
+                free_capacity_bytes: free_capacity_bytes.with_label_values(&[name, "heap"]),
             },
             mmap: LgBytesRegionMetrics {
-                alloc_count: alloc_count.with_label_values(&["mmap"]),
-                alloc_capacity_bytes: alloc_capacity_bytes.with_label_values(&["mmap"]),
-                alloc_seconds: alloc_seconds.with_label_values(&["mmap"]),
-                free_count: free_count.with_label_values(&["mmap"]),
-                free_capacity_bytes: free_capacity_bytes.with_label_values(&["mmap"]),
+                alloc_count: alloc_count.with_label_values(&[name, "mmap"]),
+                alloc_capacity_bytes: alloc_capacity_bytes.with_label_values(&[name, "mmap"]),
+                free_count: free_count.with_label_values(&[name, "mmap"]),
+                free_capacity_bytes: free_capacity_bytes.with_label_values(&[name, "mmap"]),
             },
-            mmap_disabled_count: registry.register(metric!(
-                name: "mz_bytes_mmap_disabled_count",
-                help: "count alloc attempts with lgalloc disabled",
-            )),
-            mmap_error_count: registry.register(metric!(
-                name: "mz_bytes_mmap_error_count",
-                help: "count of errors when attempting file-based mapped alloc",
-            )),
-            len_sizes: registry.register(metric!(
-                name: "mz_bytes_alloc_len_sizes",
-                help: "histogram of LgBytes alloc len sizes",
-                buckets: crate::stats::HISTOGRAM_BYTE_BUCKETS.to_vec(),
-            )),
+            alloc_seconds: alloc_seconds.with_label_values(&[name]),
+            mmap_disabled_count: mmap_disabled_count.clone(),
+            mmap_error_count: mmap_error_count.clone(),
+            len_sizes: len_sizes.clone(),
+        };
+        LgBytesMetrics {
+            persist_s3: op("persist_s3"),
+            persist_arrow: op("persist_arrow"),
         }
     }
+}
 
+impl LgBytesOpMetrics {
     /// Attempts to copy the given buf into an lgalloc managed file-based mapped
     /// region, falling back to a heap allocation.
     pub fn try_mmap<T: AsRef<[u8]>>(&self, buf: T) -> LgBytes {
+        let region = self.try_mmap_region(buf);
+        LgBytes {
+            offset: 0,
+            region: Arc::new(region),
+        }
+    }
+
+    /// Attempts to copy the given buf into an lgalloc managed file-based mapped
+    /// region, falling back to a heap allocation.
+    pub fn try_mmap_region<T: Copy>(&self, buf: impl AsRef<[T]>) -> MetricsRegion<T> {
```

Comment:
Any specific reason for the two different methods?

Reply:
You mean `try_mmap` and `try_mmap_region`? What's the alternative? Just have the `MetricsRegion<T>` one and then add something like an `impl From<MetricsRegion<u8>> for LgBytes`?